docs/Installation/QuickStartWithDockerAndVSCode.md

```diff
@@ -0,0 +1,567 @@
+    `spectre_vscode` in the option `--name spectre_vscode`.
+
+After running this command, you'll get a command prompt inside the container,
+running as root. Enter the following commands:
+
+~~~~
+# Create the directory where you will build spectre
+mkdir /work/spectre-build-clang
+
+# Install extra python packages in your container
+apt-get install python3-pip
+pip3 install autopep8 pylint jupyter notebook scipy numpy matplotlib
+pip install autopep8 pylint jupyter notebook scipy numpy matplotlib
+
+exit
+~~~~
+
+## Configure a Visual Studio Code workspace for your container
+
+### Set up extensions in the container
+
+Open Visual Studio Code. Click the docker icon on the left (looks like the
+docker menubar icon while with boxes on top), and right-click your docker
+container in the list at the top left, and start it (equivalent to `docker start
+spectre_vscode` in terminal).
+
+In Visual Studio Code, do `Shift+Command+P`, and run the command `Remote
+containers: Attach to running container...` and choose the SpECTRE container
+(named `spectre_vscode` in this guide, though you're free to change that
+name).
+
+Go to the "extensions" tab, and for each of the following extensions, install
+them in the container. You must install these extensions in the container, even
+though you already installed them in your local copy of Visual Studio Code.
+  * C/C++
+  * Format Modified
+  * GitHub Pull Requests
+  * Live Share
+  * Python
+
+You'll need to quit and re-open Visual Studio Code for these plugins to take
+effect inside the container.
+
+### Add the SpECTRE source directory to your workspace
+
+After re-opening Visual Studio Code, run the `Remote containers: Attach to
+running container...` again, if it doesn't automatically return you to a window
+connected to your container. (It should say `Container
+sxscollaboration/spectrebuildenv:latest` in the lower left corner if it is
+connected to your Docker container.)
+
+The first time you connect to your container, you'll need to open the SpECTRE
+source code's root folder. Click the button to add a folder to the workspace, or
+equivalently, un the command `Workspaces: Add Folder to Workspace`. Choose the
+path to your spectre clone (`/Users/geoffrey/Codes/spectre/spectre-alt` in the
+example command above).
+
+### Add a keyboard shortcut for git-clang-format
+
+The command `Format modified` runs git-clang-format. Choose
+`Code->Preferences->Keyboard shortcuts` (macOS) or
+open the command pallete (`Shift+Command+P` on Mac, `Shift+Control+P`
+otherwise) and find the command to modify keyboard shortcuts
+or keybindings. Then, search for `Format Modified: Format modified sections`,
+click under keybinding", and press a keyboard shortcut
+(we suggest `control+command+x`).
+
+Then every time you are editing code, press this key command to format modified
+sections (before committing) with `git clang format`.
+
+### Add .vscode configuration files
```

Comment:
We can place these in the Docker container. In particular, I think we only need `settings.json`, `cmake-kits.json` and `c_cpp_properties.json` for now, where the latter just forwards to the cmake configuration (using the `configurationProvider` and `compileCommands` options). Then all the build targets and tests are available through the cmake extension and you don't have to manually write tasks. I can send you example config files if you like.

Reply:
Actually I just copy my example config files here for completeness:

- `.vscode/settings.json`
```json
{
    "remote.containers.workspaceMountConsistency": "delegated",
    "cmake.buildDirectory": "/work/spectre-${buildKit}-${buildType}",
    "cmake.configureOnOpen": true,
    "cmake.configureArgs": [
        "-DCHARM_ROOT=/work/charm/multicore-linux64-${env.CHARM_CC}",
        "-DBUILD_PYTHON_BINDINGS=ON"
    ],
    "cmake.copyCompileCommands": "/work/vscode/compile_commands.json",
}
```

- `.vscode/cmake-kits.json`
```json
[
    {
        "name": "Clang-8",
        "compilers": {
            "CC": "/usr/bin/clang-8",
            "CXX": "/usr/bin/clang++-8",
            "Fortran": "/usr/bin/gfortran-8"
        },
        "environmentVariables": {
            "CHARM_CC": "clang"
        }
    }
]
```

- `.vscode/c_cpp_properties.json`
```json
{
    "configurations": [
        {
            "name": "Linux",
            "cStandard": "c11",
            "cppStandard": "c++17",
            "configurationProvider": "ms-vscode.cmake-tools",
            "compileCommands": "/work/vscode/compile_commands.json"
        }
    ],
    "version": 4
}
```